### DISCUSSION TITLE
Custom soundfonts?

### DISCUSSION THREAD

**Discussion by HypnoHypno** [cifkao/html-midi-player#75] [Q&A] [answered]
I'd like to use Microsoft GS Wavetable Synth soundfont for the audio, but I can't find out how the soundfont files are structured for me to do that.

**Comment by cifkao** [comment, marked as answer]
This is answered in the [FAQ](https://github.com/cifkao/html-midi-player?tab=readme-ov-file#how-can-i-use-custom-samples):
> ### How can I use custom samples?
> The player supports "SoundFonts" in a [special format](https://github.com/magenta/magenta-js/blob/d8a76682abb0979b985e4b80f6b68b5123b9f8d5/music/src/core/soundfont.ts#L381-L419) designed by Magenta. If you want to use a .sf2 file, it will not work out of the box, but it is possible to convert it with some effort. See [this discussion thread](https://github.com/cifkao/html-midi-player/discussions/43) and especially [this answer](https://github.com/cifkao/html-midi-player/discussions/43#discussioncomment-5439676), which proposes a conversion script.
